### PR TITLE
Fix #1668: SarifLogger doesn't handle sub-rules.

### LINKED ISSUE #1668: SarifLogger does not handle sub-rules
Given a `Result` object `result` with `result.RuleId == "TEST0001/1"`, and a `ReportingDescriptor` object `rule` with `rule.Id == "TEST0001"`, the call
```
SarifLogger logger = ... ;
...
logger.Log(rule, result);
```
fails with the message:
```
Error: System.ArgumentException: The rule id 'TEST0001/1' specified by the result does not
match the actual id of the rule 'TEST0001'
```

### PR DESCRIPTION
**Pull request by lgolding**

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@harleenkohli is added to the review. #Closed

**Comment by lgolding** [comment]
@rtaket is added to the review. #Closed

File: src/Sarif/ExtensionMethods.cs [hunk 1]
```diff
@@ -80,6 +80,23 @@ public static string InstanceIdLogicalComponent(this RunAutomationDetails runAut
             return instanceId.Substring(0, instanceId.LastIndexOf('/'));
         }
 
+        public static bool IsEqualToOrHierarchicalDescendantOf(this string child, string parent)
+        {
+            if (child == parent) { return true; }
+
+            string[] childComponents = child.Split(new[] { SarifConstants.HierarchicalComponentSeparator });
```
Comment (rtaket) [inline src/Sarif/ExtensionMethods.cs]: >new [](start = 51, length = 3)

Is the new[] syntax required? I thought the parameter was a params array...

Comment (lgolding) [reply src/Sarif/ExtensionMethods.cs]: Thanks! I really disliked how ugly that call was :-)

---
In reply to: [321820925](https://github.com/microsoft/sarif-sdk/pull/1669#discussion_r321820925) [](ancestors = 321820925)

**Pull request merged**